Fixed connection status icon at page load.

MMaiero opened the pull request:
Now, when the page is loaded, the correct icon is displayed, in order to provide
the right feedback on the connection status.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

cdealti commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/denali.java:
```diff
@@ -106,14 +106,14 @@ public void onSuccess(ArrayList<GwtGroupedNVPair> results) {
                                 for (GwtGroupedNVPair pair : pairs) {
                                     String name = pair.getName();
-                                    if (name != null && name.equals("kura.have.net.admin")) {
+                                    if (name != null && "kura.have.net.admin".equals(name)) {
```
null check not needed

cdealti commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/denali.java:
```diff
@@ -53,7 +53,7 @@
     private final EntryClassUi binder = GWT.create(EntryClassUi.class);
 
     private boolean isDevelopMode = false;
```
can you remove this unneeded initializer?

cdealti commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/denali.java:
```diff
@@ -140,30 +140,18 @@ public void onFailure(Throwable caught) {
-                                        denali.this.m_connected = false;
+                                        denali.this.connected = false;
                                         int connectionNameIndex = 0;
                                         for (GwtGroupedNVPair result : pairs) {
```
Since you need and increment the index in the for loop I suggest using the for loop with the index: ``for(int connectionNameIndex...``

cdealti commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/denali.java:
```diff
@@ -140,30 +140,18 @@ public void onFailure(Throwable caught) {
- // status
-                                                if ("Connection Status".equals(connectionStatus.getName())
+                                                    && result.getValue().endsWith("CloudService")) {
+                                                // done based on the idea that in the pairs data connection name is
```
``// based on the assumption the Service Status pair follows the Connection Name pair``

cdealti commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/denali.java:
```diff
@@ -140,30 +140,18 @@ public void onFailure(Throwable caught) {
+                                                // done based on the idea that in the pairs data connection name is
+                                                // before connection status
+                                                GwtGroupedNVPair connectionStatus = pairs.get(connectionNameIndex + 1);
+
```
check connectionNameIndex + 1 is less than the array length. Just in case...

cdealti commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/denali.java:
```diff
@@ -140,30 +140,18 @@ public void onFailure(Throwable caught) {
                                                         && "CONNECTED".equals(connectionStatus.getValue())) {
-                                                    denali.this.m_connected = true;
+                                                    denali.this.connected = true;
                                                     break;
```
``break`` should be moved in the outer ``if`` as soon as the Connection Name is matched

cdealti commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/Status/StatusPanelUi.java:
```diff
@@ -165,59 +165,52 @@ public void onSuccess(GwtXSRFToken token) {
+
+                        int connectionNameIndex = 0;
+
+                        for (GwtGroupedNVPair resultPair : result) {
```
same as above

cdealti commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/denali.java:
```diff
@@ -140,45 +140,35 @@ public void onFailure(Throwable caught) {
- // connection
- // status
-                                                if ("Connection Status".equals(connectionStatus.getName())
+                                                    && result.getValue().endsWith("CloudService")) {
```
I feel an iterator is more natural (snippet untested)
```
Iterator<GwtGroupedNVPair> it = pairs.iterator();
while (it.hasNext()) {
	GwtGroupedNVPair connectionName = it.next();
        if ("Connection Name".equals(connectionName.getName())
            && connectionName.getValue().endsWith("CloudService")) {

            // based on the assumption that in the ArrayList, "Service Status"
            // immediately follows "Connection Name"
            
            if (it.hasNext()) {
                GwtGroupedNVPair connectionStatus = it.next();
                if ("Service Status".equals(connectionStatus.getName())
                    && "CONNECTED".equals(connectionStatus.getValue())) {
                    denali.this.connected = true;
                }

            }

            // always break on matching the connection name
            break;
        }
}
```

MMaiero committed:
Fixed connection status icon at page load.

Now, when the page is loaded, the correct icon is displayed, in order to provide
the right feedback on the connection status.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>
